### DISCUSSION TITLE
Add option to disable --graph in git-log

### DISCUSSION THREAD

**Discussion by DamienCassou** [magit/magit#5088] [Ideas]
Some repositories I work on have a meaningless git commit graph:
![image](https://github.com/magit/magit/assets/217543/3188a0d1-776e-43a6-9944-d82862966a9a)

The graph takes valuable horizontal space and there is nothing anyone can learn from the graph (except that the git repository is a mess :smile:). It would be better to just not render it for such repositories.

Currently, the "--graph" option is passed in all cases by this code:

```emacs
(put 'magit-log-mode 'magit-log-default-arguments
     '("--graph" "-n256" "--decorate"))
```

**Comment by tarsius** [comment]
> Currently, the "--graph" option is passed in all cases by this code:

Only when using a very loose definition of "in all cases". In order to not use `--graph` by default enter the log menu with <kbd>l</kbd>, toggle the argument with <kbd>- g</kbd>, and safe that as the default using <kbd>C-x C-s</kbd>.

It would be nice if one could set different defaults for different repositories, but that isn't possible yet. It's planned though: https://github.com/magit/magit/issues/2993. That has been open for a few years now and I still don't plan to work on it *next*, but I am getting closer. (Transient improved a lot over the last half year or so, and currently I am bringing Forge a big step forward. Magit is next up for another round of improvements. (Though this particular part, will mostly happen in Transient.))

**Reply by DamienCassou** [reply]
> Only when using a very loose definition of "in all cases".

indeed. I wrote that too quickly. Thanks for clarifying.

**Comment by tarsius** [comment]
dc6074f8daa391dc6556ae5097c9a02afd136ba3 adds Git v2.55's new `--graph-lane-limit` to log menus.  It is hidden by default and not available when using an older Git.

**Reply by DamienCassou** [reply]
Thank you very much
